aws-android-sdk-iot-test/src/androidTest/java/com/amazonaws/mobileconnectors/iot/IotClient.java

```diff
@@ -0,0 +1,279 @@
+    }
+
+    // Deletes any associated certificates, and detaches/deletes the policy.
+    void cleanupPolicy(String policyName) {
+        Log.d(TAG, "Cleaning up policy " + policyName + ". Will remove any/all certificates/documents.");
+        List<String> certificateArns = listTargetsForPolicy(policyName);
+        for (String certificateArn : certificateArns) {
+            detachPolicy(policyName, certificateArn);
+        }
+        for (String certificateId : getCertificateIdsFromArns(certificateArns)) {
+            deleteCertificate(certificateId);
+        }
+        deletePolicy(policyName);
+    }
+
+    List<String> listTargetsForPolicy(String policyName) {
+        Log.d(TAG, "Listing all targets for policy named = " + policyName);
+        ListTargetsForPolicyRequest request = new ListTargetsForPolicyRequest();
+        request.setPolicyName(policyName);
+        ListTargetsForPolicyResult result;
+        try {
+            result = iot.listTargetsForPolicy(request);
+        } catch (ResourceNotFoundException noSuchPolicyException) {
+            return Collections.emptyList();
+        }
+        return result.getTargets();
+    }
+
+    private void createPolicy(String policyName, String policyDocument) {
+        Log.d(TAG, "Creating policy " + policyName + " with document = " + policyDocument);
+        CreatePolicyRequest createPolicyRequest;
+        createPolicyRequest = new CreatePolicyRequest();
+        createPolicyRequest.setPolicyName(policyName);
+        createPolicyRequest.setPolicyDocument(policyDocument);
+        iot.createPolicy(createPolicyRequest);
+    }
+
+    private void attachPolicy(String policyName, String certificateArn) {
+        Log.d(TAG, "Attaching policy " + policyName + " from certificate ARN = " + certificateArn);
+        AttachPolicyRequest attachPolicyRequest = new AttachPolicyRequest();
+        attachPolicyRequest.setPolicyName(policyName);
+        attachPolicyRequest.setTarget(certificateArn);
+        iot.attachPolicy(attachPolicyRequest);
+    }
+
+    void detachPolicy(@NonNull String policyName, @Nullable String certificateArn) {
+        Objects.requireNonNull(policyName);
+        Log.d(TAG, "Detaching policy " + policyName + " from certificate ARN = " + certificateArn);
+        DetachPolicyRequest detachPolicyRequest = new DetachPolicyRequest();
+        detachPolicyRequest.setPolicyName(policyName);
+        if (certificateArn != null) {
+            detachPolicyRequest.setTarget(certificateArn);
+        }
+        iot.detachPolicy(detachPolicyRequest);
+    }
+
+    private void deletePolicy(String policyName) {
+        Log.d(TAG, "Deleting policy: " + policyName);
+        DeletePolicyRequest deletePolicyRequest = new DeletePolicyRequest();
+        deletePolicyRequest.setPolicyName(policyName);
+        try {
+            iot.deletePolicy(deletePolicyRequest);
+        } catch (ResourceNotFoundException noSuchPolicy) {
+            // Hey, cool. It doesn't exist.
```

Comment:
In a test run context, I'd expect to have this boom bubble up. Or is this part of an `@AfterClass` so we want to prevent that? Either way, perhaps a `Log.d` at the very least?

Reply:
It is in test teardown, yea. I guess it is more the teardown's prerogative to ignore the deletion failure. I can move the catch over there.